src/base/RecoveryCodeCollection.php

```diff
@@ -0,0 +1,128 @@
+<?php
+declare(strict_types=1);
+
+namespace hiqdev\yii2\mfa\base;
+
+use Yii;
+use yii\base\BaseObject;
+
+class RecoveryCodeCollection extends BaseObject
+{
+    protected array $codes = [];
+
+    protected int $count = 10;
+
+    protected int $blocks = 4;
+
+    protected int $blockLength = 3;
+
+    protected string $blockSeparator = '-';
+
+    public function getCodes(): array
+    {
+        return $this->codes;
+    }
+
+    public function setCount(int $count): self
+    {
+        $this->count = $count;
+
+        return $this;
+    }
+
+    public function getCount(): int
+    {
+        return $this->count;
+    }
+
+    public function setBlocks(int $blocks): self
+    {
+        $this->blocks = $blocks;
+
+        return $this;
```

Comment:
FYI:
I dont really like the approach of fluent interfaces, if you dont use or need them. And I am not alone: https://ocramius.github.io/blog/fluent-interfaces-are-evil/

Reply:
changed locally, push later